central/graphql/resolvers/node_vulnerabilities.go

```diff
@@ -80,8 +141,239 @@ func (resolver *Resolver) NodeVulnCounter(ctx context.Context, args RawQuery) (*
+		return nil, err
+	}
+	query = tryUnsuppressedQuery(query)
+	fixableVulnsQuery := search.ConjunctionQuery(query, search.NewQueryBuilder().AddBools(search.Fixable, true).ProtoQuery())
+	fixableVulns, err := vulnLoader.FromQuery(ctx, fixableVulnsQuery)
+	if err != nil {
+		return nil, err
+	}
+
+	unFixableVulnsQuery := search.ConjunctionQuery(query, search.NewQueryBuilder().AddBools(search.Fixable, false).ProtoQuery())
+	unFixableCVEs, err := vulnLoader.FromQuery(ctx, unFixableVulnsQuery)
+	if err != nil {
+		return nil, err
+	}
+
+	return mapNodeCVEsToVulnerabilityCounter(fixableVulns, unFixableCVEs), nil
+}
+
+// CreatedAt is the time a node CVE first seen in the system
+func (resolver *nodeCVEResolver) CreatedAt(ctx context.Context) (*graphql.Time, error) {
+	value := resolver.data.GetCveBaseInfo().GetCreatedAt()
+	return timestamp(value)
+}
+
+// CVE name of the node CVE
+func (resolver *nodeCVEResolver) CVE(ctx context.Context) string {
+	return resolver.data.GetCveBaseInfo().GetCve()
+}
+
+// EnvImpact is the fraction of nodes that contain the nodeCVE
+func (resolver *nodeCVEResolver) EnvImpact(ctx context.Context) (float64, error) {
+	nodeLoader, err := loaders.GetNodeLoader(ctx)
+	if err != nil {
+		return 0, err
+	}
+	numNodes, err := nodeLoader.CountAll(ctx)
+	if err != nil || numNodes == 0 {
+		return 0, err
+	}
+
+	query := resolver.getNodeCVEQuery()
+	numNodesWithCVE, err := nodeLoader.CountFromQuery(ctx, query)
+	if err != nil || numNodesWithCVE == 0 {
+		return 0, err
+	}
+	return float64(numNodesWithCVE) / float64(numNodes), nil
+}
+
+// FixedByVersion returns the version of the parent component that removes this CVE
+func (resolver *nodeCVEResolver) FixedByVersion(_ context.Context) (string, error) {
+	scope, hasScope := scoped.GetScope(resolver.ctx)
+	if !hasScope {
+		return "", nil
+	}
+	if scope.Level != v1.SearchCategory_NODE_COMPONENTS {
+		return "", nil
+	}
+	edgeID := edges.EdgeID{ParentID: scope.ID, ChildID: resolver.data.GetId()}.ToString()
+	edge, _, err := resolver.root.NodeComponentCVEEdgeDataStore.Get(resolver.ctx, edgeID)
+	if err != nil {
+		return "", err
+	}
+	return edge.GetFixedBy(), nil
+}
+
+// ID of the node CVE
+func (resolver *nodeCVEResolver) ID(ctx context.Context) graphql.ID {
+	value := resolver.data.GetId()
+	return graphql.ID(value)
+}
+
+// IsFixable returns whether node CVE is fixable by any component
+func (resolver *nodeCVEResolver) IsFixable(ctx context.Context, args RawQuery) (bool, error) {
+	query, err := args.AsV1QueryOrEmpty(search.ExcludeFieldLabel(search.CVE))
+	if err != nil {
+		return false, err
+	}
+
+	scope, ok := scoped.GetScope(resolver.ctx)
+	if !ok || scope.Level != v1.SearchCategory_NODE_VULNERABILITIES {
+		query = search.ConjunctionQuery(query, resolver.getNodeCVEQuery())
+	}
+
+	query = search.ConjunctionQuery(query, search.NewQueryBuilder().AddBools(search.Fixable, true).ProtoQuery())
+	count, err := resolver.root.NodeComponentCVEEdgeDataStore.Count(ctx, query)
```

Comment:
You can fire the query on node CVE datastore since the result of inner join would be the same. That way we can remove dependency from nodecomponent-cve edge datastore in graphQL.